LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  TherapySettingsView.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 7/7/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import AVFoundation
+import HealthKit
+import LoopKit
+import SwiftUI
+
+public protocol TherapySettingsViewDelegate: class {
+    func gotoEdit(therapySetting: TherapySetting)
+    func save()
+    func cancel()
+}
+
+public class TherapySettingsViewModel: ObservableObject {
+    private var initialTherapySettings: TherapySettings
+    var therapySettings: TherapySettings
+
+    public init(therapySettings: TherapySettings = preview_therapySettings) {
+        self.therapySettings = therapySettings
+        self.initialTherapySettings = therapySettings
+    }
+    
+    /// Reset to original
+    func reset() {
+        therapySettings = initialTherapySettings
+    }
+}
+
+public struct TherapySettingsView: View, HorizontalSizeClassOverride {
+    @Environment(\.dismiss) var dismiss
+
+    weak var delegate: TherapySettingsViewDelegate?
+    
+    @ObservedObject var viewModel: TherapySettingsViewModel
+
+    @State var isEditing: Bool = false
+    
+    // Hack to test edit button below section or next to header
+    public enum DesignVersion {
+        case A, B, C
+        func cycle() -> DesignVersion {
+            switch self {
+            case .A: return .B
+            case .B: return .C
+            case .C: return .A
+            }
+        }
+    }
+    @State var editButtonDesignVersion: DesignVersion = .C
+
+    public enum PresentationMode {
```

Comment:
There's also an enum named `PresentationMode` in LoopKit (in addition to a SwiftUI environmental variable called `presentationMode`). Your changes built fine for me, but I'm wondering if it would be less confusing if the `TherapySettingsView` enum had a different name. Perhaps `ConfigurationMode`?

Reply:
I'll have to think about that one. I notice the other version has modal and flow, but with no description of what the two versions really mean. Otherwise, I might just use that.

Reply:
`modal` is intended for being presented as a therapy setting (as it's presented 'modally'), and `flow` is for being presented as part of the prescription flow. Sorry about not commenting that in, I'll do that on my next PR. You could 100% use that here

Reply:
Ahhhhh ok.  Done.